ckanapi/common.py

```diff
@@ -40,7 +40,7 @@ def __getattr__(self, name):
         def action(**kwargs):
             files = {}
             for k, v in kwargs.items():
-                if hasattr(v, 'read'):
+                if hasattr(v, 'read') or (isinstance(v, (list, tuple)) and len(v) >= 2 and hasattr(v[1], 'read')):
```

Comment:
I think we should only allow tuples to avoid confusion with the JSON list type that could be passed. Also, why do you need to accept sequences with more than two elements? What would it mean to have a file name, file object then other values trailing?